stixcore/products/level0/science.py

```diff
@@ -84,12 +85,115 @@ def to_requests(self):
                              ssid=self.ssid, control=control, data=data)
 
 
+class RawPixelData(ScienceProduct):
+    def __init__(self, *, service_type, service_subtype, ssid, control, data, **kwargs):
+        super().__init__(service_type=service_type, service_subtype=service_subtype,
+                         ssid=ssid, control=control, data=data, **kwargs)
+        self.name = 'xray-rpd'
+        self.level = 'L0'
+
+    @classmethod
+    def from_levelb(cls, levelb):
+        packets = BaseProduct.from_levelb(levelb)
+
+        service_type = packets.get('service_type')[0]
+        service_subtype = packets.get('service_subtype')[0]
+        ssid = packets.get('pi1_val')[0]
+
+        control = ControlSci.from_packets(packets)
+
+        # control.remove_column('num_structures')
+        control = unique(control)
+
+        if len(control) != 1:
+            raise ValueError('Creating a science product form packets from multiple products')
+
+        control['index'] = 0
+
+        data = Data()
+        data['start_time'] = (np.array(packets.get_value('NIX00404'), np.uint16)) * 0.1 * u.s
+        data['rcr'] = np.array(packets.get_value('NIX00401', attr='value'), np.ubyte)
+        data['integration_time'] = (np.array(packets.get_value('NIX00405'), np.int16)) * 0.1 * u.s
+        data['pixel_masks'] = _get_pixel_mask(packets, 'NIXD0407')
+        data['detector_masks'] = _get_detector_mask(packets)
+        data['triggers'] = np.array([packets.get_value(f'NIX00{i}') for i in range(408, 424)],
+                                    np.int64).T
+        data['num_samples'] = np.array(packets.get_value('NIX00406'), np.int16)
+
+        num_detectors = 32
+        num_energies = 32
+        num_pixels = 12
+
+        # Data
+        tmp = dict()
+        tmp['pixel_id'] = np.array(packets.get_value('NIXD0158'), np.ubyte)
+        tmp['detector_id'] = np.array(packets.get_value('NIXD0153'), np.ubyte)
+        tmp['channel'] = np.array(packets.get_value('NIXD0154'), np.ubyte)
+        tmp['continuation_bits'] = np.array(packets.get_value('NIXD0159'), np.ubyte)
```

Comment:
the data['...'].meta entries are missing

Reply:
Created a new issue for this.